src/CrosschainDeployScript.sol

```diff
@@ -0,0 +1,191 @@
+        public
+    {
+        NetworkIds memory deploymentTargetNetworkIds = _stringToNetworkIds[deploymentTarget];
+        uint8 deploymentTargetDomainId = deploymentTargetNetworkIds.InternalDomainId;
+        require(deploymentTargetDomainId != 0, "Invalid deployment target");
+        _domainIds.push(deploymentTargetDomainId);
+        uint256 deploymentTargetChainId = deploymentTargetNetworkIds.ChainId;
+        _chainIds.push(deploymentTargetChainId);
+        _constructorArgs.push(constructorArgs);
+        _initDatas.push(initData);
+    }
+
+    /**
+     * @notice this function takes in the contract string, in the form that
+     * @notice `forge`'s `getCode` takes it, along with some other parameters and passes
+     * @notice it along to the `deploy` function of the `CrossChainDeployAdapter`
+     * @notice contract.
+     * @param gasLimit Contract deploy and init gas.
+     * @param isUniquePerChain True to have unique addresses on every chain.
+     *   Users call this function and pass only the function call string as
+     *   `MyContract.sol:MyContract`. The function call string is then parsed
+     *   and the `callData` and `bytesCode` are extracted from it.
+     *   and the contract is deployed on the other chains.
+     */
+    function deploy(uint256 gasLimit, bool isUniquePerChain)
+        public
+        payable
+        hasDeploymentNetworks
+        returns (address[] memory)
+    {
+        // We use the contractString to get the bytecode of the contract,
+        // reference: https://book.getfoundry.sh/cheatcodes/get-code
+        bytes memory deployByteCode = vm.getCode(contractString);
+        bytes32 salt;
+        if (_staticSalt == 0x00000000000000000000) {
+            salt = generateSalt();
+        } else {
+            salt = _staticSalt;
+        }
+
+        uint256[] memory fees = ICrosschainDeployAdapter(crosschainDeployContractAddress).calculateDeployFee(
+            deployByteCode, gasLimit, salt, isUniquePerChain, _constructorArgs, _initDatas, _domainIds
+        );
+        uint256 totalFee;
+        uint256 feesArrayLength = fees.length;
+        for (uint256 j = 0; j < feesArrayLength; j++) {
+            uint256 fee = fees[j];
+            totalFee += fee;
+        }
+        // TODO: Would I need to check if totalFee < msg.value since this `deploy` is itself payable and called with some fee?
+        ICrosschainDeployAdapter(crosschainDeployContractAddress).deploy{value: totalFee}(
+            deployByteCode, gasLimit, salt, isUniquePerChain, _constructorArgs, _initDatas, _domainIds, fees
+        );
+        address[] memory contractAddresses = new address[](_chainIds.length);
+        for (uint256 k = 0; k < _chainIds.length; k++) {
+            address contractAddress = ICrosschainDeployAdapter(crosschainDeployContractAddress)
+                .computeContractAddressForChain(msg.sender, salt, isUniquePerChain, _chainIds[k]);
```

Comment:
https://github.com/foundry-rs/foundry/issues/3885 according to this msg.sender will probably not work as expected. Need to find a way to specify the actual deployer wallet address.